### ISSUE TITLE
[Android] Does Fury support usage on Android?

### ISSUE THREAD

**Issue by Kaneki-x** [apache/fory#1101]
## Is your feature request related to a problem? Please describe.
After integrating into the Android project, some problems were discovered. The project probably did not adapt to the Java environment under Android, and there would be exceptions when running. Currently, it seems that _JDKAccess needs to be adapted.

## Describe the solution you'd like
Hope to increase the support and adaptation for serialization capabilities under the Android platform.

## Additional context
<img width="716" alt="image" src="https://github.com/alipay/fury/assets/6857108/5ba2a857-105f-4144-a387-91f3da68b4c9">
<img width="863" alt="image" src="https://github.com/alipay/fury/assets/6857108/5b4d8525-927c-4098-a0a8-ff72ee4b59df">

**Comment by chaokunyang** [comment]
Some issues may need to be addressed for andriod support
- Doest StringSerializer/StringBufferSerializer/StringBuilder needs to be adpated? Fury check `value` type by `char[]/byte[]`. Seems work for android too.
- Fury needs JDK8+, JDK8 API(Stream/Lambda) are used by fury. is this supported for all andriod versions? If not, lower  andriod versions needs to fallback to other frameworks
- Fury will generate bytecode dynamically using janino and load it as class. I'm not sure whether this work for andriod?

If all above issues can be addressed for andriod, I think fury will work well for andriod too.

**Comment by chaokunyang** [comment]
Another issue is that you'd better create Fury with `FuryBuilder#withAsyncCompilation(true)` to avoid compaliation latency. You can't warmup for clients.

**Comment by neboskreb** [comment]
There is an architectural problem which makes support for Android cumbersome: the "bundled" serializers are installed too early, before the builder even has a chance to control this. 

To remedy this, the instantiation and warm-up of the serializers needs to be postponed enough so the user could install platform-specific serializers which they deem proper. Once this is unblocked, there is no more need to tweak the bundled `StringSerializer`/`StringBufferSerializer`/`StringBuilderSerializer` as the user will provide those specific to their platform (popular platforms like Android can be supported by Fury by offering e.g. `AndroidStringSerializer`)
